ts/store/actions/messages.ts

```diff
@@ -141,14 +142,39 @@ export const resetMigrationStatus = createAction(
   "MESSAGES_MIGRATE_TO_PAGINATED_DONE"
 );
 
-export type MessagesActions =
-  | ActionType<typeof reloadAllMessages>
-  | ActionType<typeof loadNextPageMessages>
-  | ActionType<typeof loadPreviousPageMessages>
-  | ActionType<typeof loadMessageDetails>
-  | ActionType<typeof migrateToPaginatedMessages>
-  | ActionType<typeof resetMigrationStatus>
-  | ActionType<typeof removeMessages>
-  | ActionType<typeof upsertMessageStatusAttributes>
-  | ActionType<typeof loadMessageById>
-  | ActionType<typeof loadThirdPartyMessage>;
+/**
+ * The user requests an attachment download.
+ */
+export const downloadAttachment = createAsyncAction(
+  "DOWNLOAD_ATTACHMENT_REQUEST",
+  "DOWNLOAD_ATTACHMENT_SUCCESS",
+  "DOWNLOAD_ATTACHMENT_FAILURE",
+  "DOWNLOAD_ATTACHMENT_CANCEL"
+)<
+  UIAttachment,
+  { attachment: UIAttachment; path: string },
```

Comment:
Should we use the `Download` type defined in `ts/store/reducers/entities/messages/download.ts` here?